python_bindings/src/halide/imageio.py

```diff
@@ -0,0 +1,73 @@
+try:
+    import imageio.v2 as imageio
+except:
+    import imageio
+import numpy
+
+
+def is_interleaved(im):
+    """If the given ndarray is 3-dimensional and appears to have an interleaved
+       layout, return True. Otherwise, return False."""
+
+    # Assume that 'interleaved' will only apply to channels <= 4
+    return im.ndim == 3 and im.strides[2] == 1 and im.shape[2] in [1, 2, 3, 4]
+
+
+def _as_interleaved(im):
+    """If the given ndarray is 3-dimensional and appears to be planar layout,
+       return a copy that is in interleaved form, leaving the input unchanged.
+       Otherwise, return the image ndarray unchanged.
+       Note that this call must be used with care, as the returnee may or may
+       not be a copy."""
+    if im.ndim == 3 and not is_interleaved(im):
+        return numpy.moveaxis(im, 0, 2)
+    else:
+        return im
+
+
+def _as_planar(im):
+    """If the given ndarray is 3-dimensional and appears to be interleaved
+       layout, return a copy that is in planar form, leaving the input
```

Comment:
I don't think this comment is right. There's no copy here.

Reply:
oops. fixed.